template/node/Cargo.toml

```diff
@@ -18,39 +18,39 @@ log = "0.4.8"
 structopt = "0.3.8"
 jsonrpc-core = "14.0.3"
 
-sp-api = { version = "2.0.0-dev", path = "../../vendor/substrate/primitives/api" }
-sp-blockchain = { version = "2.0.0-dev", path = "../../vendor/substrate/primitives/blockchain" }
-sc-rpc-api = { version = "0.8.0-dev", path = "../../vendor/substrate/client/rpc-api" }
-sc-rpc = { version = "2.0.0-dev", path = "../../vendor/substrate/client/rpc" }
-substrate-frame-rpc-system = { version = "2.0.0-dev", path = "../../vendor/substrate/utils/frame/rpc/system" }
-pallet-transaction-payment-rpc = { version = "2.0.0-dev", path = "../../vendor/substrate/frame/transaction-payment/rpc/" }
-sc-cli = { version = "0.8.0-dev", path = "../../vendor/substrate/client/cli" }
-sp-core = { version = "2.0.0-dev", path = "../../vendor/substrate/primitives/core" }
-sc-executor = { version = "0.8.0-dev", path = "../../vendor/substrate/client/executor" }
-sc-service = { version = "0.8.0-dev", path = "../../vendor/substrate/client/service" }
-sp-inherents = { version = "2.0.0-dev", path = "../../vendor/substrate/primitives/inherents" }
-sc-transaction-pool = { version = "2.0.0-dev", path = "../../vendor/substrate/client/transaction-pool" }
-sp-transaction-pool = { version = "2.0.0-dev", path = "../../vendor/substrate/primitives/transaction-pool" }
-sc-network = { version = "0.8.0-dev", path = "../../vendor/substrate/client/network" }
-sc-consensus-aura = { version = "0.8.0-dev", path = "../../vendor/substrate/client/consensus/aura" }
-sp-consensus-aura = { version = "0.8.0-dev", path = "../../vendor/substrate/primitives/consensus/aura" }
-sc-consensus-manual-seal = { path = "../../vendor/substrate/client/consensus/manual-seal" }
-sp-consensus = { version = "0.8.0-dev", path = "../../vendor/substrate/primitives/consensus/common" }
-sc-consensus = { version = "0.8.0-dev", path = "../../vendor/substrate/client/consensus/common" }
-sp-timestamp = { version = "2.0.0-dev", default-features = false, path = "../../vendor/substrate/primitives/timestamp" }
-evm = { version = "2.0.0-dev", package = "pallet-evm", path = "../../vendor/substrate/frame/evm" }
+sp-api = { version = "2.0.0-dev", git = "https://github.com/paritytech/substrate.git", rev = "8a35a14c2dd91bc5de77366f45a06fe45bdfd1d6" }
```

Comment:
Maybe just `branch = "master"` and pin commit in `Cargo.lock`?

Reply:
It will be hard to make sure the pinned version doesn't get updated when adding new deps

Reply:
https://github.com/paritytech/polkadot repo does use this strategy though. Another worry for me is how do we get this large amount of dependency pins updated each time.

Reply:
Global search & replace is easy enough. 

Polkadot can constantly keep up to date because Substrate PR much have companion Polkadot PR as well and unless frontier have same treatment, I don’t think pin to master branch is good idea.

Reply:
I agree with @sorpaas that updating the revision number is inconvenient, specially when switching from multiple remotes and versions during development.

Reply:
During development, you can use local version of substrate instead by update your cargo config, see read me update in this PR

Reply:
I will prefer to pin to a branch instead of master, similar how cumulus is currently doing. But someone will need to setup that branch in Substrate first.

Reply:
@xlc I created `frontier` branch in Substrate. Please update this PR to use that.